ts/WoltLabSuite/Core/Ui/Poll/Manager/Vote.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Handles the poll voting.
+ *
+ * @author  Joshua Ruesweg
+ * @copyright  2001-2022 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Poll/Manager/Vote
+ * @since   5.5
+ */
+
+import AjaxRequest from "../../../Ajax/Request";
+import Manager, { PollViews } from "./Manager";
+import { ResponseData } from "../../../Ajax/Data";
+import * as Core from "../../../Core";
+
+export class Vote {
+  private pollManager: Manager;
+
+  private button: HTMLButtonElement;
+
+  private inputs: NodeListOf<HTMLInputElement>;
```

Comment:
I would use an array instead, it isn't a live list anyway so there is no value in not converting it to an array right away.